File: notification-schema-java/src/main/java/uk/gov/defra/tracesx/notificationschema/representation/PostalAddress.java
package uk.gov.defra.tracesx.notificationschema.representation;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import javax.validation.constraints.NotBlank;
import lombok.Builder;
import lombok.Data;
import lombok.extern.jackson.Jacksonized;

@Builder
@Data
@JsonInclude(Include.NON_EMPTY)
@Jacksonized
public class PostalAddress {

  @NotBlank
  private String addressLine1;
  private String addressLine2;
  private String county;
  @NotBlank
  private String cityOrTown;
  @NotBlank
  private String postalCode;
}

File: imports-frontend-entities/src/entities/postal_address.js
const handler = require('./base/handler')

module.exports = class PostalAddress {

  constructor(obj = {}) {

    this.addressLine1 = obj.addressLine1
    this.addressLine2 = obj.addressLine2
    this.addressLine3 = obj.addressLine3
    this.addressLine4 = obj.addressLine4
    this.county = obj.county
    this.cityOrTown = obj.cityOrTown
    this.postalCode = obj.postalCode
    return Object.seal(new Proxy(this, handler))
  }
}
